test/test_default_args.py

```diff
@@ -81,3 +82,103 @@ class SomeClass:
         parser.parse_args(shlex.split(f"--a {passed_value}"))
     with raises_missing_required_arg():
         parser.parse_args("")
+
+
+def test_suppress_default(simple_attribute, silent):
+    some_type, passed_value, expected_value = simple_attribute
+
+    @dataclass
+    class SomeClass:
+        a: some_type
+        b: int = 5
+
+    parser = ArgumentParser()
+    parser.add_arguments(SomeClass, dest='some_class', default=argparse.SUPPRESS)
+
+    parsed_args, unparsed_args = parser.parse_known_args(shlex.split(f'--a {passed_value} --b 0'))
+    assert parsed_args == argparse.Namespace(some_class={'a': expected_value, 'b': 0})
+    assert unparsed_args == []
+
+    parsed_args, unparsed_args = parser.parse_known_args(shlex.split(f'--a {passed_value}'))
+    assert parsed_args == argparse.Namespace(some_class={'a': expected_value})
+    assert unparsed_args == []
+
+    with raises_missing_required_arg():
+        parser.parse_args('')
+
+    with raises_missing_required_arg():
+        parser.parse_args(shlex.split('--b 0'))
+
+
+def test_suppress_with_regular_args(simple_attribute, silent):
+    some_type, passed_value, expected_value = simple_attribute
+
+    @dataclass
+    class SomeClass:
+        a: some_type
+        b: int = 5
+
+    parser = ArgumentParser()
+    parser.add_arguments(SomeClass, dest='some_class', default=argparse.SUPPRESS)
+    if some_type == bool:
+        def strbool(s):
+            if s.lower() in ['false', 'no', 'f']:
+                return False
+            else:
+                return True
+
+        parser.add_argument('--c', type=strbool, default=argparse.SUPPRESS)
+    else:
+        parser.add_argument('--c', type=some_type, default=argparse.SUPPRESS)
```

Comment:
Can you mix suppressed and non-suppressed arguments? What happens in that case?

Reply:
I've added another test case to demonstrate this